viahtml/hooks/hooks.py

```diff
@@ -124,9 +147,19 @@ def modify_tag_attrs(self, tag, attrs):
         if tag == "link" and ("rel", "canonical") in attrs:
             stop = True
 
-        ## Allow iframes to hint to Via to not proxy
-        if tag == "iframe" and ("data-viahtml-no-proxy", None) in attrs:
-            stop = True
+        if tag == "iframe":
+            # Allow individual iframes to disable proxying via an attribute.
+            if ("data-viahtml-no-proxy", None) in attrs:
+                stop = True
+
+            # Disable proxying for all frames if `via.proxy_frames` query param was set.
+            #
+            # Iframe proxying defaults to true for backwards compatibility.
+            proxy_frames = query_param_as_bool(
```

Comment:
I could've been easier to introduce the setting as a negative, eg. `disable_proxy_frames` but maybe you are thinking ahead of this being the default behavior in the future.

Reply:
I just find it slightly more straightforward to reason about the outcome of `feature=on|off` vs `disable_feature=true|false`. Eventually I'd indeed prefer to minimize the amount of iframes that we proxy by default though. We'll start with preventing proxying in limited contexts, and then we can roll it out more widely.